assignments/assignments.go

```diff
@@ -49,6 +52,13 @@ func UpdateFromTestsRepo(logger *zap.SugaredLogger, db database.Database, course
 		return
 	}
 	logger.Debugf("Assignments for %s successfully updated from '%s' repo", course.GetCode(), pb.TestsRepo)
+
+	ctx := context.Background()
```

Comment:
Move this `ctx` line early in this method; above `fetchAssignments` and use it also in `fetchAssignments` instead of `context.Background()`. We typically use the same context for all calls that we may need to cancel or timeout... although we don't do that here at the moment.